tiny_cnn/config.h

```diff
@@ -69,6 +69,16 @@ namespace tiny_cnn {
  * you can change it to float, or user defined class (fixed point,etc)
  **/
 typedef float float_t;
+typedef long long int64;
+typedef int int32;
+typedef short int16;
+typedef signed char int8;
+typedef int qint32;
+typedef short qint16;
+typedef signed char qint8;
+typedef unsigned int quint32;
+typedef unsigned short quint16;
+typedef unsigned char  quint8;
```

Comment:
If you need fixed size types, you should use the ones from `<cstdint>` ([link](http://en.cppreference.com/w/cpp/header/cstdint)), I believe that applies here too. It can otherwise create quite a havoc on certain platforms.